spotify/src/test/java/org/wso2/carbon/connector/integration/test/common/ConnectorIntegrationUtil.java

```diff
@@ -0,0 +1,371 @@
+            while ((len = response.read(bytes)) != -1) {
+                sb.append(new String(bytes, 0, len));
+            }
+
+            if (!sb.toString().trim().isEmpty()) {
+                out = sb.toString();
+            }
+        }
+
+        JSONObject jsonObject = new JSONObject(out);
+
+        return jsonObject;
+    }
+
+    public static OMElement sendXMLRequest(String addUrl, String query) throws MalformedURLException, IOException,
+            XMLStreamException {
+
+        String charset = "UTF-8";
+        URLConnection connection = new URL(addUrl).openConnection();
+        connection.setDoOutput(true);
+        connection.setRequestProperty("Accept-Charset", charset);
+        connection.setRequestProperty("Content-Type", "application/json;charset=" + charset);
+        OutputStream output = null;
+        try {
+            output = connection.getOutputStream();
+            output.write(query.getBytes(charset));
+        } finally {
+            if (output != null) {
+                try {
+                    output.close();
+                } catch (IOException logOrIgnore) {
+                    log.error("Error while closing the connection");
+                }
+            }
+        }
+
+        HttpURLConnection httpConn = (HttpURLConnection) connection;
+        InputStream response;
+
+        if (httpConn.getResponseCode() >= 400) {
+            response = httpConn.getErrorStream();
+        } else {
+            response = connection.getInputStream();
+        }
+
+        String out = "{}";
+        if (response != null) {
+            StringBuilder sb = new StringBuilder();
+            byte[] bytes = new byte[1024];
+            int len;
+            while ((len = response.read(bytes)) != -1) {
+                sb.append(new String(bytes, 0, len));
+            }
+
+            if (!sb.toString().trim().isEmpty()) {
+                out = sb.toString();
+            }
+        }
+
+        OMElement omElement = AXIOMUtil.stringToOM(out);
+
+        return omElement;
+
+    }
+
+    public static Properties getConnectorConfigProperties(String connectorName) {
+
+        String connectorConfigFile = null;
+        ProductConstant.init();
+        try {
+            connectorConfigFile =
+                    ProductConstant.SYSTEM_TEST_SETTINGS_LOCATION + File.separator + "artifacts" + File.separator
+                            + "ESB" + File.separator + "connector" + File.separator + "config" + File.separator
+                            + connectorName + ".properties";
+            File connectorPropertyFile = new File(connectorConfigFile);
+            InputStream inputStream = null;
+            if (connectorPropertyFile.exists()) {
+                inputStream = new FileInputStream(connectorPropertyFile);
+            }
+
+            if (inputStream != null) {
+                Properties prop = new Properties();
+                prop.load(inputStream);
+                inputStream.close();
```

Comment:
inputStream not closed in a finally block which can lead to potential memory leaks!